src/main/messageExtractor/JSONExtractor.java

```diff
@@ -0,0 +1,31 @@
+package messageExtractor;
+
+import org.json.simple.parser.JSONParser;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.ParseException;
+
+public class JSONExtractor implements MessageExtractorInterface{
+    @Override
+    public PeripheralAction handleMessage(String input) throws UnexpectedActionException {
```

Comment:
parseMessage might be a better name, as 'handle' is a bit misleading: it could mean anything.